### PR TITLE
Position caret at end after uiEntrySetText() on Windows, replicating …

### PR DESCRIPTION
**Pull request by adamierymenko**
…behavior on MacOS

Just that. A few lines.

### CODE REVIEW

**Review by cody271 — CHANGES_REQUESTED** [review]
Same failure in CI due to odd formatting/encoding with this PR as well:
```
../windows/entry.cpp: In function 'void uiEntrySetText(uiEntry*, const char*)':
../windows/entry.cpp:69:5: error: '\U000000a0' was not declared in this scope
   69 |         int l = (int)strlen(text);
      |     ^
../windows/entry.cpp:70:2: error: expected ';' before '\U000000a0'
   70 |         if (l >= 0) {
      |  ^~
      |  ;
```

File: windows/entry.cpp [hunk 1]
```diff
@@ -66,6 +66,10 @@ void uiEntrySetText(uiEntry *e, const char *text)
 	// doing this raises an EN_CHANGED
 	e->inhibitChanged = TRUE;
 	uiWindowsSetWindowText(e->hwnd, text);
+        int l = (int)strlen(text);
+        if (l >= 0) {
```
Comment (szanni) [inline windows/entry.cpp:70]: Unless I am missing something: how would this condition ever be `false`? `strlen` returns `size_t`, so it is by definition always `>=0`.
I would generally prefer matching types too `size_t len = strlen(text);` and then use that.

**Review by szanni — CHANGES_REQUESTED** [review]

File: windows/entry.cpp (hunk 1 above)
Comment (szanni) [reply windows/entry.cpp:70]: Oh and variable declaration at the top of the function. To stay consistent,

**Comment by adamierymenko** [comment]
Fixed but didn't make it size_t because I think Windows expects int.

File: windows/entry.cpp [hunk 2]
```diff
@@ -63,9 +63,12 @@ char *uiEntryText(uiEntry *e)
 
 void uiEntrySetText(uiEntry *e, const char *text)
 {
+	int l;
 	// doing this raises an EN_CHANGED
 	e->inhibitChanged = TRUE;
 	uiWindowsSetWindowText(e->hwnd, text);
+        l = (int)strlen(text);
```
Comment (szanni) [inline windows/entry.cpp:70]: Indentation is one tab, not spaces. You can see the mismatch in the Github diff.

**Review by szanni — CHANGES_REQUESTED** [review]

**Comment by szanni** [comment]
> Fixed but didn't make it size_t because I think Windows expects int.

Yeah, well windows actually expects a signed and unsigned type. Great. `UINT_PTR` and `LONG_PTR` according to `WPARAM` and `LPARAM` as defined [here](https://docs.microsoft.com/en-us/windows/win32/winprog/windows-data-types).
I guess `int` is fine. It somehow does not seem to matter.

Otherwise the code does what it is supposed to. Unix behavior is matching too,

File: windows/entry.cpp [hunk 3]
```diff
@@ -63,10 +63,15 @@ char *uiEntryText(uiEntry *e)
...
 void uiEntrySetText(uiEntry *e, const char *text)
 {
+	int l;
+	if (!text)
+		text = "";
 	// doing this raises an EN_CHANGED
 	e->inhibitChanged = TRUE;
-	uiWindowsSetWindowText(e->hwnd, text);
-	e->inhibitChanged = FALSE;
+        uiWindowsSetWindowText(e->hwnd, text);
```
Comment (szanni) [inline windows/entry.cpp:71]: I am not sure what editor you are using, but it seems incapable of getting the indentation right?

**Review by szanni — CHANGES_REQUESTED** [review]

File: windows/entry.cpp (hunk 3 above)
Comment (adamierymenko) [reply windows/entry.cpp:71]: I was using the GitHub online web editor to make small PRs. Will probably ditch that and actually check out a fork and do it in a proper editor. I guess the online editor is broken and for some reason inserts pseudo-whitespace unicode.

Comment (szanni) [reply windows/entry.cpp:71]: Oh wow. That sucks. Seems to work here (Firefox 101 on Linux).

**Review by szanni — CHANGES_REQUESTED** [review]
Yup, good. The white space/indentation issues are still there though. L69-73.

**Review by cody271 — APPROVED** [review]
Confirmed to be passing CI.. hopefully no more GitHub online editor formatting weirdness. 😕 

I will just fix manually in master after merge. LGTM.

**Comment by szanni** [comment]
Sorry for opening this issue again: @adamierymenko what was your exact use case here or the problematic behavior?

Is my understanding correct that this was about the caret placement when the `uiEntry` already has focus? To place the caret at the end when focused?

Because if the `uiEntry` does not have the focus the caret is always placed where I click. Correct?

Thanks

Related to #232 and #233

**Pull request merged**